google/cloud/aiplatform/metadata/context.py

```diff
@@ -57,3 +58,82 @@ def _update_resource(
         cls, client: utils.MetadataClientWithOverride, resource: proto.Message,
     ) -> proto.Message:
         return client.update_context(context=resource)
+
+    @classmethod
+    def add_artifacts_or_executions(
+        cls,
+        context_id: str,
+        artifact_ids: Optional[Sequence[str]] = None,
```

Comment:
For these methods that take nodes as parameters, it's much more preferable to:
1. Pass in the object representations so we are not duplicating the resource naming code at every call and can just rely on the constructor of the class.
2. They should be instance methods instead of class methods so we are not duplicating the api client instantiation at every call.

It seems like that would reduce the amount of code needed for these methods and avoid the requirement for passing in project/location/creds at every call.

The source in this method would reduce to:

```python
self.api_client.add_context_artifacts_and_executions(
            context=self.resource_name,
            artifacts=[node.resource_name for node in artifacts],
            executions=[node.resource_name for node in executions],
        )
```

Reply:
Thanks for the suggestion. I updated this method and the one in exeuction.py to instance methods. I did changed the input artifacts/executions to full resource str instead of the full object as the full resource name should be all we need to call the backend API and we would already have that if we have the resource object.

Also updated metadata.py and related tests.